src/components/VDataTable/mixins/body.js

```diff
@@ -36,17 +36,23 @@ export default {
       }
 
       const rows = []
-      for (let index = 0, len = this.filteredItems.length; index < len; ++index) {
+      for (let index = 0; index < this.filteredItems.length; ++index) {
         const item = this.filteredItems[index]
         const props = this.createProps(item, index)
-        const row = this.$scopedSlots.items(props)
+        let row = this.$scopedSlots.items(props)
 
-        rows.push(this.needsTR(row)
-          ? this.genTR(row, {
+        if (this.needsTR(row)) {
+          row = this.genTR(row, {
             key: index,
             attrs: { active: this.isSelected(item) }
           })
-          : row)
+        } else {
+          for (let childIndex = 0; childIndex < row.length; ++childIndex) {
+            const rowChild = row[childIndex]
+            rowChild.key = rowChild.key || `${index}:${childIndex}`
+          }
+        }
+        rows.push(row)
```

Comment:
If you want speedy fast, `rows` should be initialised to the length you need, and use `rows[i]` instead of push.